server/util/retry/retry.go

```diff
@@ -236,7 +237,7 @@ func Do[T any](ctx context.Context, opts *Options, fn func(ctx context.Context)
 			return rsp, nil
 		}
 		if _, ok := err.(*nonRetryableError); ok {
-			logFailedAttempt(lastError, " and could not be retried due to a non-retryable error")
+			logFailedAttempt(err, fmt.Sprintf(" (last error: %s) and could not be retried due to a non-retryable error", lastError))
```

Comment:
do we need to log `lastError`? seems like this was a bug and we should be just logging `err` instead of `lastErr` here.